#! /bin/sh /usr/share/dpatch/dpatch-run
## 16fix_gcc_warnings_II.dpatch
##
## DP: ...

@DPATCH@
diff -Naur screen-4.0.3.orig/display.c screen-4.0.3/display.c
--- screen-4.0.3.orig/display.c	2008-06-01 23:39:48.000000000 +0200
+++ screen-4.0.3/display.c	2008-06-01 23:41:19.000000000 +0200
@@ -4423,7 +4423,7 @@
     }
 }
 
-struct lay *
+struct layout *
 CreateLayout(title, startat)
 char *title;
 int startat;
diff -Naur screen-4.0.3.orig/extern.h screen-4.0.3/extern.h
--- screen-4.0.3.orig/extern.h	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/extern.h	2008-06-01 23:41:19.000000000 +0200
@@ -289,6 +289,10 @@
 #endif
 extern void  SetCanvasWindow __P((struct canvas *, struct win *));
 extern int   MakeDefaultCanvas __P((void));
+extern void  RecreateCanvasChain __P((void));
+extern int   CountCanvasPerp __P((struct canvas *));
+extern void  EqualizeCanvas __P((struct canvas *, int));
+extern void  ResizeCanvas __P((struct canvas *));
 extern int   AddCanvas __P((int));
 extern void  RemCanvas __P((void));
 extern void  OneCanvas __P((void));
@@ -307,6 +311,12 @@
 extern void  ResetIdle __P((void));
 extern void  KillBlanker __P((void));
 extern void  DisplaySleep1000 __P((int, int));
+extern void  SaveLayout __P((char *, struct canvas *));
+extern void  AutosaveLayout __P((struct layout *));
+extern struct layout *FindLayout __P((char *));
+extern void  LoadLayout __P((struct layout *, struct canvas *));
+extern void  NewLayout __P((char *, int));
+extern void  ShowLayouts __P((int));
 
 /* resize.c */
 extern int   ChangeWindowSize __P((struct win *, int, int, int));
diff -Naur screen-4.0.3.orig/help.c screen-4.0.3/help.c
--- screen-4.0.3.orig/help.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/help.c	2008-06-01 23:41:19.000000000 +0200
@@ -1302,7 +1302,7 @@
 	      if (wtab[pos] && WTAB_GROUP_MATCHES(pos))
 		break;
 	  if (pos == -1)
-	    pos == MAXWIN;
+	    pos = MAXWIN;
 	}
     }
   wlistdata->pos = pos;
diff -Naur screen-4.0.3.orig/input.c screen-4.0.3/input.c
--- screen-4.0.3.orig/input.c	2008-06-01 23:40:40.000000000 +0200
+++ screen-4.0.3/input.c	2008-06-01 23:41:19.000000000 +0200
@@ -142,7 +142,7 @@
   inpdata->inpmode = mode;
   inpdata->privdata = data;
   if (!priv)
-    priv = &inpdata->privdata;
+    priv = (char *)&inpdata->privdata;
   inpdata->priv = priv;
   inpdata->inpstringlen = 0;
   inpdata->inpstring = NULL;
diff -Naur screen-4.0.3.orig/layer.c screen-4.0.3/layer.c
--- screen-4.0.3.orig/layer.c	2008-06-01 23:41:11.000000000 +0200
+++ screen-4.0.3/layer.c	2008-06-01 23:41:19.000000000 +0200
@@ -398,7 +398,6 @@
 {
   struct canvas *cv;
   struct viewport *vp;
-  char *s2;
   int xs2, xe2, y2, len, len2;
   struct mchar or;
 
diff -Naur screen-4.0.3.orig/process.c screen-4.0.3/process.c
--- screen-4.0.3.orig/process.c	2008-06-01 23:41:11.000000000 +0200
+++ screen-4.0.3/process.c	2008-06-01 23:41:19.000000000 +0200
@@ -2070,15 +2070,15 @@
 #endif
     case RC_WINDOWLIST:
       if (!*args)
-        display_wlist(0, WLIST_NUM, (char *)0);
+        display_wlist(0, WLIST_NUM, (struct win *)0);
       else if (!strcmp(*args, "-m") && !args[1])
-        display_wlist(0, WLIST_MRU, (char *)0);
+        display_wlist(0, WLIST_MRU, (struct win *)0);
       else if (!strcmp(*args, "-b") && !args[1])
-        display_wlist(1, WLIST_NUM, (char *)0);
+        display_wlist(1, WLIST_NUM, (struct win *)0);
       else if (!strcmp(*args, "-b") && !strcmp(args[1], "-m") && !args[2])
-        display_wlist(1, WLIST_MRU, (char *)0);
+        display_wlist(1, WLIST_MRU, (struct win *)0);
       else if (!strcmp(*args, "-m") && !strcmp(args[1], "-b") && !args[2])
-        display_wlist(1, WLIST_MRU, (char *)0);
+        display_wlist(1, WLIST_MRU, (struct win *)0);
       else if (!strcmp(*args, "string"))
 	{
 	  if (args[1])
@@ -3967,7 +3967,7 @@
 	  if (args[0][0])
 	    {
 	      fore->w_group = WindowByName(*args);
-	      if (fore->w_group && fore->w_group != W_TYPE_GROUP)
+	      if (fore->w_group && fore->w_type != W_TYPE_GROUP)
 		fore->w_group = 0;
 	    }
 	}
@@ -4071,7 +4071,6 @@
 	}
       else if (!strcmp(args[0], "select"))
 	{
-          struct layout *lay;
           if (!args[1])
 	    {
 	      Input("Switch to layout: ", 20, INP_COOKED, SelectLayoutFin, NULL, 0);
@@ -4952,7 +4951,7 @@
 {
   register struct win **pp;
   int n = fore ? fore->w_number : MAXWIN;
-  char *group = fore ? fore->w_group : 0;
+  struct win *group = fore ? fore->w_group : (struct win *)0;
 
   for (pp = fore ? wtab + n + 1 : wtab; pp != wtab + n; pp++)
     {
@@ -4974,7 +4973,7 @@
 {
   register struct win **pp;
   int n = fore ? fore->w_number : -1;
-  char *group = fore ? fore->w_group : 0;
+  struct win *group = fore ? fore->w_group : (struct win *)0;
 
   for (pp = wtab + n - 1; pp != wtab + n; pp--)
     {
@@ -5485,14 +5484,13 @@
 int len;
 char *data;	/* dummy */
 {
-  int n;
   struct layout *lay;
 
   if (!len || !display)
     return;
   if (len == 1 && *buf == '-')
     {
-      LoadLayout((struct layout *)0);
+      LoadLayout((struct layout *)0, (struct canvas *)0);
       Activate(0);
       return;
     }
@@ -6445,7 +6443,7 @@
 int flags;
 {
   struct canvas *cv;
-  int nreg, dsize, diff, siz, nsiz, l, done;
+  int diff, l;
   int gflag = 0, abs = 0, percent = 0;
   int orient = 0;
 
diff -Naur screen-4.0.3.orig/resize.c screen-4.0.3/resize.c
--- screen-4.0.3.orig/resize.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/resize.c	2008-06-01 23:41:19.000000000 +0200
@@ -146,9 +146,8 @@
 int change_fore;
 {
   struct win *p;
-  struct canvas *cv, **cvpp;
+  struct canvas *cv;
   int wwi;
-  int y, h, hn, xe, ye;
 
   debug2("ChangeScreenSize from (%d,%d) ", D_width, D_height);
   debug3("to (%d,%d) (change_fore: %d)\n",wi, he, change_fore);
diff -Naur screen-4.0.3.orig/socket.c screen-4.0.3/socket.c
--- screen-4.0.3.orig/socket.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/socket.c	2008-06-01 23:41:19.000000000 +0200
@@ -1193,7 +1193,6 @@
   char *p;
   int pid;
   int noshowwin;
-  int r = 0;
   struct win *wi;
 
   ASSERT(display);
@@ -1318,7 +1317,7 @@
 #endif
 	{
 	  flayer = D_forecv->c_layer;
-	  display_wlist(1, WLIST_NUM, (char *)0);
+	  display_wlist(1, WLIST_NUM, (struct win *)0);
 	  noshowwin = 1;
 	}
     }
